### ISSUE TITLE
Build on NPM doesn't work

### ISSUE THREAD

**Issue by rtransat** [team-innovation/vue-sfc-rollup#33]
I have run the build with the `npm run build` command and it's working good.
I've published the package on [NPM](https://www.npmjs.com/package/vhue) and when I try to use it I have the following error: 

```
[Vue warn]: Error in render: "TypeError: _vm._ssrNode is not a function"
__vue_render__@https://zq0n5.csb.app/node_modules/vhue/dist/vhue.ssr.js:4440:66
```

Why it's use the ssr version of the package?
When I install the package with [yalc](https://github.com/whitecolor/yalc) on local it's works.

You can try it on [codesandbox](https://codesandbox.io/s/vhue-zq0n5) if you want to help.

**Comment by mgdodge** [comment]
I suspect the problem is in how codesandbox is configured. Modern bundlers should see `dist/vhue.esm.js` and that includes webpack as configured by the vue cli. You should only see the ssr build when running in node (eg. nuxt or other ssr renderer.) Reading their docs, it seems they use a customized verison of webpack, etc.

Things seem to work fine in the vue cli for rapid development, as well as the `npm run build` script generated by the cli. I even went so far as to test in nuxt via `npx create-nuxt-app` to run it in a true SSR context, and the ssr worked fine, as well as the client-side rehydration that nuxt does.

I think this needs to be taken to codesandbox as an issue for them to resolve. They are doing something different in how they choose which file to serve from package.json.

**Comment by rtransat** [comment]
Thank for your answer.

Maybe you can update your script to have what [Vue.js](https://fr.vuejs.org/v2/cookbook/packaging-sfc-for-npm.html#How-do-I-make-multiple-versions-of-my-component) offers?

The main use umd in our example and you use the ssr, the ssr need to be in other folder (if we following our example)

That could fix the issue #31 

What do you think?

**Comment by mgdodge** [comment]
Could you give me an example of what you are saying with the Vue.js offering?

Putting the umd build into `main` instead of the ssr build could be considered a 'breaking' change, so I will not be doing that without ample evidence that it is the right way to go.

I currently have [a pull request](https://github.com/vuejs/vuejs.org/pull/2172) pending to update vuejs.org with the following information:

The ssr build in the `main` field is done because that is where nuxt.js, quasar, and other ssr applications can look for things. They don't look at the `module` field currently because node.js doesn't, and they are node apps, after all. The ssr-optimized build should go there so all the ssr optimizations like string concatenation, etc. will be there. Other bundlers will pick up the `module` field just fine, so the esm build goes there.

This is the basis for putting the ssr build where it is. The issue with codesandbox is the only time it has come up as problematic. I still think that their customized version of webpack is where the proper fix is, since the standard version of webpack, with the exact same files, works flawlessly.

**Comment by mgdodge** [comment]
As per discussion on the [related codesandbox issue](https://github.com/codesandbox/codesandbox-client/issues/3305), I just published v2.3.0 that sets the `browser` field to the same build as the `module` field, effectively fixing things in codesandbox and other bundlers that don't see the `module` field. Webpack, nuxt, quasar, and others in my testing should be unaffected.

**Comment by rtransat** [comment]
Thx you for your work.

You can check the demo of my component [here](https://zq0n5.csb.app/) ([source](https://codesandbox.io/s/vhue-zq0n5))

Thank again ;)

PS: If you know a good boilerplate with rollup for building a react component library it will be great 👍
